File: lib/api/landingPage.js
var _ = require("lodash"),
  Promise = require("bluebird"),
  util = require("../util"),
  qs = require("querystring"),
  log = util.logger();

function LandingPage(marketo, connection) {
  this._marketo = marketo;
  this._connection = connection;
}

LandingPage.prototype = {
  get: function (id, draft) {
    var path = util.createAssetPath(`landingPage/${id}.json`);
    if (draft) {
      path += `?status=draft`;
    }

    var options = _.extend(
      {},
      {
        _method: "GET",
      }
    );
    return this._connection.get(path, { data: options });
  },
  approveDraft: function (id) {
    var path = util.createAssetPath(`landingPage/${id}/approveDraft.json`);

    var options = _.extend(
      {},
      {
        _method: "POST",
      }
    );
    return this._connection.post(path, { data: options });
  },
  getLandingPages: function (options) {
    var path = util.createAssetPath("landingPages.json");
    options = _.extend({}, options, {
      _method: "GET",
    });
    return this._connection.get(path, { data: options });
  },
};

module.exports = LandingPage;

File: src.net/Marketo-Rest/Api/LandingPage.cs
using System.Threading.Tasks;

namespace Marketo.Api
{
    /// <summary>
    /// Class LandingPage.
    /// </summary>
    public class LandingPage
    {
        readonly MarketoClient _marketo;
        readonly Connection _connection;

        /// <summary>
        /// Initializes a new instance of the <see cref="LandingPage"/> class.
        /// </summary>
        /// <param name="marketo">The marketo.</param>
        /// <param name="connection">The connection.</param>
        public LandingPage(MarketoClient marketo, Connection connection)
        {
            _marketo = marketo;
            _connection = connection;
        }

        /// <summary>
        /// Gets the landing pages.
        /// </summary>
        /// <param name="options">The options.</param>
        /// <returns>Task&lt;dynamic&gt;.</returns>
        public Task<dynamic> GetLandingPages(dynamic options = null)
        {
            var path = util.createAssetPath("landingPages.json");
            return _connection.get(path, new { data = options });
        }
    }
}
